src/ORM/DataObject.php

```diff
@@ -3422,7 +3422,13 @@ public static function get_one($callerClass = null, $filter = "", $cache = true,
 
         $item = null;
         if (!$cache || !isset(self::$_cache_get_one[$callerClass][$cacheKey])) {
-            $dl = DataObject::get($callerClass)->where($filter)->sort($orderBy);
+            $dl = DataObject::get($callerClass);
+            if (!empty($filter)) {
+                $dl = $dl->where($filter);
+            }
+            if (!empty($orderBy)) {
+                $dl = $dl->sort($orderBy);
```

Comment:
Since this is using sort explicitly here, we should update the PHPDoc (and probably param name) which imply order by (and thus raw SQL) can be used.

Reply:
Updated

Reply:
Updated